TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/PATCH-Positive-DeviceCommand.robot

```diff
@@ -4,41 +4,52 @@ Resource     TAF/testCaseModules/keywords/core-metadata/coreMetadataAPI.robot
-    And deviceCommand Should Be Updated
-    [Teardown]  Delete Device Profile By Name
+    And Profile Test-Profile-1 command isHidden Should Be Updated
+    [Teardown]  Delete Device Profile By Name  Test-Profile-1
 
-ProfileCommandPATCH002 - Update multiple deviceCommands on one device profile
-    Given Create a device profile with multiple deviceCommands
-    When Update deviceCommand
+ProfileCommandPATCH002 - Update multiple Command on multiple device profiles
+    # multiple commands > multiple profiles
+    Given Generate Multiple Device Profiles Sample
+    And Create Device Profile ${deviceProfile}
+    And Generate multiple command samples for updating
+    When Update command ${commandUpdate}
     Then Should Return Status Code "207"
     And Should Return Content-Type "application/json"
     And Item Index All Should Contain Status Code "200"
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And deviceCommand Should Be Updated
-    [Teardown]  Delete Device Profile By Name
-
-ProfileCommandPATCH003 - Update multiple deviceCommand on multiple device profiles
-    Given Create multiple device profiles with multiple deviceCommands
-    When Update deviceCommand
-    Then Should Return Status Code "207"
-    And Should Return Content-Type "application/json"
-    And Item Index All Should Contain Status Code "200"
-    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
-    And deviceCommand Should Be Updated
-     [Teardown]  Delete Multiple Device Profiles By Names
+    And Profile Test-Profile-1 command isHidden Should Be Updated
+    And Profile Test-Profile-2 command isHidden Should Be Updated
+    And Profile Test-Profile-3 command isHidden Should Be Updated
+    [Teardown]  Delete Multiple Device Profiles By Names  Test-Profile-1  Test-Profile-2  Test-Profile-3
 
+*** Keywords ***
+Generate multiple command samples for updating
+    ${command_1}=  Create Dictionary  name=CurrentStatus  isHidden=${false}
+    ${command_2}=  Create Dictionary  name=MotorKind   isHidden=${true}
+    ${command_3}=  Create Dictionary  name=CurrentHumidity  isHidden=${true}
+    Generate updating command  ${command_1}  ${command_2}  ${command_3}
+    Set To Dictionary  ${commandUpdate}[0]  profileName=Test-Profile-1
+    Set To Dictionary  ${commandUpdate}[1]  profileName=Test-Profile-2
+    Set To Dictionary  ${commandUpdate}[2]  profileName=Test-Profile-3
 
+Profile ${profile_name} command isHidden Should Be Updated
+    Query device profile by name  ${profile_name}
+    Run Keyword IF  "${profile_name}" == "Test-Profile-1"
+    ...             Should Be Equal  ${content}[profile][deviceCommands][0][isHidden]  ${false}
+    ...    ELSE IF  "${profile_name}" == "Test-Profile-2"  Should Be Equal  ${content}[profile][deviceCommands][0][isHidden]  ${true}
+    ...    ELSE IF  "${profile_name}" == "Test-Profile-3"  Should Be Equal  ${content}[profile][deviceCommands][0][isHidden]  ${true}
```

Comment:
`if else` is unnecessary in this case if `true/false` is a variable in the keyword name
```suggestion
Profile ${profile_name} command isHidden Should Be ${bool}
    Query device profile by name  ${profile_name}
    Should Be Equal  ${content}[profile][deviceCommands][0][isHidden]  ${bool}
```

Reply:
updated